presentation/src/main/java/me/androidbox/presentation/agenda/viewmodel/AgendaViewModel.kt

```diff
@@ -130,4 +144,29 @@ class AgendaViewModel @Inject constructor(
             }
         }
     }
+
+    private fun syncAgendaItems() {
+        agendaSyncObserver = Observer { workInfo ->
+            if(workInfo.state == WorkInfo.State.SUCCEEDED) {
+                /* Delete the ids from the deleted sync table */
+                viewModelScope.launch {
+                    agendaLocalRepository.deleteEventSyncType(SyncAgendaType.DELETE)
+                }
+            }
+        }
+
+        val syncAgendaId = viewModelScope.async {
+            syncAgendaItems.sync()
+        }
+
+        viewModelScope.launch {
+            workManagerLiveData = workManager.getWorkInfoByIdLiveData(syncAgendaId.await())
+            workManagerLiveData.observeForever(agendaSyncObserver)
+        }
+    }
```

Comment:
I think there could be many issues with this `syncAgendaItems`  Please provide some code snippets to help provide some clarity.

Reply:
@MKiperszmid and @philipplackner Just for the sack of learning. Is my `syncAgendaItems` correct. If not how can I make this correct? 

I have already made the changes as requested by @MKiperszmid to pass in the dao to the worker. But just interested to know if I am doing something wrong with the above code.

Reply:
I personally only work with flows and converted the LiveData to a Flow (I think you need a dependency for that). Then did something like this to listen for the result after I called enqueue():

<img width="1174" alt="Bildschirmfoto 2023-04-17 um 22 01 45" src="https://user-images.githubusercontent.com/53933333/232491745-2a2fc7d1-f9e7-42d3-8a92-2ab14f30be83.png">